acme/dns_challenge.go

```diff
@@ -155,12 +157,28 @@ func dnsQuery(fqdn string, rtype uint16, nameserver string, recursive bool) (in
 		m.RecursionDesired = false
 	}
 
-	in, err = dns.Exchange(m, nameserver)
-	if err == dns.ErrTruncated {
-		tcp := &dns.Client{Net: "tcp"}
-		in, _, err = tcp.Exchange(m, nameserver)
-	}
+	// Try up to 3 times (returning early if nil errors)
+	for i := 1; i <= 3; i++ {
+		in, err = dns.Exchange(m, nameserver)
+
+		if err == dns.ErrTruncated {
+			tcp := &dns.Client{Net: "tcp"}
+			// The following should reset the err to be nil
+			in, _, err = tcp.Exchange(m, nameserver)
+		}
+
+		if err == nil {
+			return
+		}
+
+		logf("[WARNING][%s] acme: DNS Query #%d failed: %s", fqdn, i, err)
```

Comment:
A dropped UDP packet isn't really something i would warn the user about as long as we are still able to solve by retrying.